test/wasm2asm/i64-lowering.wast

```diff
@@ -0,0 +1,269 @@
+;; Testing i64 lowering.
+
+(module
+  (func $dummy)
+
+  (func (export "eq_i64") (param $0 i64) (param $1 i64) (result i32)
+    (i64.eq (get_local $0) (get_local $1)))
+
+  (func (export "ne_i64") (param $0 i64) (param $1 i64) (result i32)
+    (i64.ne (get_local $0) (get_local $1)))
+
+  (func (export "ge_s_i64") (param $0 i64) (param $1 i64) (result i32)
+    (i64.ge_s (get_local $0) (get_local $1)))
+
+  (func (export "gt_s_i64") (param $0 i64) (param $1 i64) (result i32)
+    (i64.gt_s (get_local $0) (get_local $1)))
+
+  (func (export "le_s_i64") (param $0 i64) (param $1 i64) (result i32)
+    (i64.le_s (get_local $0) (get_local $1)))
+
+  (func (export "lt_s_i64") (param $0 i64) (param $1 i64) (result i32)
+    (i64.lt_s (get_local $0) (get_local $1)))
+
+  (func (export "ge_u_i64") (param $0 i64) (param $1 i64) (result i32)
+    (i64.ge_u (get_local $0) (get_local $1)))
+
+  (func (export "gt_u_i64") (param $0 i64) (param $1 i64) (result i32)
+    (i64.gt_u (get_local $0) (get_local $1)))
+
+  (func (export "le_u_i64") (param $0 i64) (param $1 i64) (result i32)
+    (i64.le_u (get_local $0) (get_local $1)))
+
+  (func (export "lt_u_i64") (param $0 i64) (param $1 i64) (result i32)
+    (i64.lt_u (get_local $0) (get_local $1)))
+)
+
+;; TODO: change things so we can invoke with i64 args and automatically lower
+;; the arguments, rather than doing it manually.
+(assert_return (invoke "eq_i64" (i32.const 0) (i32.const 0)
+                                (i32.const 0) (i32.const 0))
+               (i32.const 1))
+(assert_return (invoke "eq_i64" (i32.const 0) (i32.const 0)
```

Comment:
remind me, are these invokes translated into js and executed in the test suite? or are they used in some other manner?

Reply:
They are translated to asm.js and executed via this code: https://github.com/WebAssembly/binaryen/blob/11a83d799d1b539f7c7a1eec987dc87119a098dd/scripts/test/wasm2asm.py#L47-L52

Reply:
I see, thanks. Looks pretty good there.

Eventually, as we discussed yesterday, it would be nice to hook up the wasm test suite (in `tests/spec/`). I think we want that anyhow, and also it might save writing some of the new i64 tests.